crates/matrix-sdk-base/src/client.rs

```diff
@@ -90,6 +91,7 @@ pub struct BaseClient {
     olm_machine: Arc<RwLock<Option<OlmMachine>>>,
     /// Observable of when a user is ignored/unignored.
     pub(crate) ignore_user_list_changes: SharedObservable<()>,
+    pub(crate) roominfo_update_sender: Arc<StdRwLock<Option<broadcast::Sender<OwnedRoomId>>>>,
```

Comment:
Could it be non-optional so we avoid deferred initialization (aka no more `set_roominfo_update_sender`), and this struct exposes a method to get a receiver, instead?

Reply:
You might want to use the client builder to fill this field without an optional value.
Also, we should have the _receiver_ in the client, and the _sender_s in the `RoomInfo`s, right? Except if you use this sender to create the receivers? Anyway, please add a documentation to explain that :-).